pinot-spi/src/main/java/org/apache/pinot/spi/stream/PartitionGroupMetadataFetcher.java

```diff
@@ -63,8 +65,8 @@ public Exception getException() {
   @Override
   public Boolean call()
       throws Exception {
-    String clientId = PartitionGroupMetadataFetcher.class.getSimpleName() + "-"
-            + _streamConfig.getTableNameWithType() + "-" + _topicName;
+    String clientId = PartitionGroupMetadataFetcher.class.getSimpleName() + "-" + _topicName + "-try" + Integer
+        .toString(++_callRetries);
```

Comment:
1. This is adding Kafka client specific code to `PartitionGroupMetadataFetcher` which is agnostic of the stream technology used.
2. If we have consumers for multiple kafka partitions on the same host then all of them would get the same clientId? Not that the duplicate clientId related MBean exceptions only get logged and don't throw, so the consumer still gets created though it is missing MBeans which could impact monitoring.
3. Do we even need this? The `AppInfoParser.registerAppInfo` call is the *last* call made in the `KafkaConsumer` constructor.

Reply:
1. They are already existing code & all stream tech ( kafka , pulsar, kinesis ) have a topic name .
2. Yes . But here the mBean would be already existing , so we would only ending up with a warning in logs 
3. This is a temporary solution . I did put up a fix in kafka for the same https://github.com/apache/kafka/pull/14287 ( this is approved ) , but not sure how soon we would upgrade the packages to have it.

Reply:
> They are already existing code & all stream tech ( kafka , pulsar, kinesis ) have a topic name .

No I mean it doesn't make sense for the clientId to be incremented on retries generally for all stream tech.

> This is a temporary solution . I did put up...

The point I am making is that this is not really needed and doesn't solve the problem anyways. If there's a MBean already with name "classname-mytopic-0", then all subsequent calls will still have the MBean error logged because retry won't kick in anyways.

If your goal is to handle intermittent issues like DNS failures then the MBean won't be registered either ways because the AppInfoParser call is the last call and wouldn't be made (the DNS failure would throw before the code reaches there)

Reply:
> If your goal is to handle intermittent issues like DNS failures then the MBean won't be registered either ways because the AppInfoParser call is the last call 

Thanks for pointing out. I am reverting the change of adding retry ids